pkg/controller/openapi/openapi_status_reconciler.go

```diff
@@ -0,0 +1,191 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OpenAPIStatusReconciler struct {
+	*reconcilers.BaseReconciler
+	resource            *capabilitiesv1beta1.OpenAPI
+	providerAccountHost string
+	reconcileError      error
+	reconcileReady      bool
+	logger              logr.Logger
+}
+
+func NewOpenAPIStatusReconciler(b *reconcilers.BaseReconciler, resource *capabilitiesv1beta1.OpenAPI, providerAccountHost string, reconcileError error, reconcileReady bool) *OpenAPIStatusReconciler {
+	return &OpenAPIStatusReconciler{
+		BaseReconciler:      b,
+		resource:            resource,
+		providerAccountHost: providerAccountHost,
+		reconcileError:      reconcileError,
+		reconcileReady:      reconcileReady,
+		logger:              b.Logger().WithValues("Status Reconciler", resource.Name),
+	}
+}
+
+func (s *OpenAPIStatusReconciler) Reconcile() (reconcile.Result, error) {
+	s.logger.V(1).Info("START")
+
+	newStatus, err := s.calculateStatus()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	equalStatus := s.resource.Status.Equals(newStatus, s.logger)
+	s.logger.V(1).Info("Status", "status is different", !equalStatus)
+	s.logger.V(1).Info("Status", "generation is different", s.resource.Generation != s.resource.Status.ObservedGeneration)
+	if equalStatus && s.resource.Generation == s.resource.Status.ObservedGeneration {
+		// Steady state
+		s.logger.V(1).Info("Status steady state, status was not updated")
+		return reconcile.Result{}, nil
+	}
+
+	// Save the generation number we acted on, otherwise we might wrongfully indicate
+	// that we've seen a spec update when we retry.
+	// TODO: This can clobber an update if we allow multiple agents to write to the
+	// same status.
+	newStatus.ObservedGeneration = s.resource.Generation
+
+	s.logger.V(1).Info("Updating Status", "sequence no:", fmt.Sprintf("sequence No: %v->%v", s.resource.Status.ObservedGeneration, newStatus.ObservedGeneration))
+
+	s.resource.Status = *newStatus
+	updateErr := s.Client().Status().Update(s.Context(), s.resource)
+	if updateErr != nil {
+		// Ignore conflicts, resource might just be outdated.
+		if errors.IsConflict(updateErr) {
+			s.logger.Info("Failed to update status: resource might just be outdated")
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		return reconcile.Result{}, fmt.Errorf("Failed to update status: %w", updateErr)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (s *OpenAPIStatusReconciler) calculateStatus() (*capabilitiesv1beta1.OpenAPIStatus, error) {
+	newStatus := &capabilitiesv1beta1.OpenAPIStatus{}
+
+	newStatus.ProviderAccountHost = s.providerAccountHost
+
+	productResourceName, err := s.getManagedProduct()
+	if err != nil {
+		return nil, err
+	}
+	newStatus.ProductResourceName = productResourceName
+
+	backendResourceNames, err := s.getManagedBackends()
```

Comment:
Can multiple backends belong to a single OpenAPI resource?
looking at the OpenAPIBackendReconciler code it seems only a Backend is created :thinking:

Reply:
What request are you referring to? Do you mean in the OpenAPI spec document?

Reply:
https://issues.redhat.com/browse/THREESCALE-6197

Reply:
:+1: